lib/js/components/Form/Checkbox/Checkbox.vue

```diff
@@ -1,60 +1,256 @@
-import { SELECTION_CONTROL_TYPE } from '../SelectionControl/SelectionControl.consts';
-
-export default defineComponent({
-	name: 'Checkbox',
-	components: { SelectionControl },
-	props: {
-		size: {
-			type: String as PropType<CheckboxSize>,
-			default: CHECKBOX_SIZE.SMALL,
-			validator(size: CheckboxSize) {
-				return Object.values(CHECKBOX_SIZE).includes(size);
-			},
-		},
-		label: {
-			type: String,
-			default: null,
-		},
-		isSelected: {
-			type: Boolean,
-			default: false,
-		},
-		state: {
-			type: String as PropType<CheckboxState>,
-			default: CHECKBOX_STATE.DEFAULT,
-			validator(state: CheckboxState) {
-				return Object.values(CHECKBOX_STATE).includes(state);
-			},
-		},
-	},
-	// TODO fix me when touching this file
-	// eslint-disable-next-line vue/require-emit-validator
-	emits: ['update:is-selected', 'input:focus', 'input:blur'],
-	data() {
-		return {
-			SELECTION_CONTROL_TYPE: Object.freeze(SELECTION_CONTROL_TYPE),
-			ICONS: Object.freeze(ICONS),
-		};
-	},
+<style scoped lang="scss">
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/animations';
+
+.ds-checkbox {
+	$root: &;
+
+	align-items: center;
+	cursor: pointer;
+	display: inline-flex;
+
+	&.-ds-x-small {
+		gap: $space-4xs;
+	}
+
+	&.-ds-small {
+		gap: $space-2xs;
+	}
+
+	&.-ds-medium {
+		gap: $space-2xs;
+	}
+
+	&__root {
+		background: none;
+		border: none;
+		line-height: 0;
+		margin: 0;
+		padding: 0;
+		position: relative;
+
+		#{$root}.-ds-x-small & {
+			padding: $space-3xs 0;
+		}
+
+		#{$root}.-ds-small & {
+			padding: $space-3xs 0;
+		}
+
+		#{$root}.-ds-medium & {
+			padding: $space-2xs 0;
+		}
+
+		&:focus {
+			outline: none;
+		}
+
+		&::before {
+			aspect-ratio: 1;
+			border-radius: $radius-xl;
+			content: '';
+			display: block;
+			height: 100%;
+			position: absolute;
+			top: 0;
+			transform: scale(0);
+			// see https://cubic-bezier.com/#.23,1,.32,1
+			transition: all $default-transition-time cubic-bezier(0.23, 1, 0.32, 1) 0ms;
+			width: auto;
+		}
+
+		#{$root}:hover:not(.-ds-disabled) &::before,
+		&:focus::before {
+			transform: scale(1);
+		}
+
+		#{$root}.-ds-x-small &::before {
+			left: -6px;
+		}
+
+		#{$root}.-ds-small &::before {
+			left: -6px;
+		}
+
+		#{$root}.-ds-medium &::before {
+			left: -8px;
+		}
+
+		#{$root}:hover &::before {
+			background-color: $color-neutral-background-ghost-hovered;
+		}
+
+		&:focus::before {
+			background-color: $color-neutral-background-ghost-focused;
+		}
+	}
+
+	&__indicator {
+		color: $color-neutral-icon;
+		position: relative;
+
+		#{$root}.-ds-error & {
+			color: $color-danger-icon;
+		}
+
+		#{$root}.-ds-disabled & {
+			color: $color-neutral-icon-disabled;
+		}
+
+		#{$root}.-ds-elevation &::before {
+			background: $color-default-background;
+			bottom: 2px;
```

Comment:
I bet, this will be a first bug in this component when we update FontAwesome in future. Making elevation size based on icon structure is very risky.

Maybe we should consider using [DuoTone](https://fontawesome.com/icons/square?f=duotone&s=regular&sc=%23ffffff&so=1) icons?